jobrunner/executors/local.py

```diff
@@ -395,22 +395,41 @@ def finalize_job(job_definition):
 
     # First get the user-friendly message for known database exit codes, for jobs
     # that have db access
-    message = None
```

Comment:
If you add a `unmatched_hint = None` here, you won't need to declare it in each branch of the if/elif/else below. In `write_log_file()` a job_metadata entry only gets written out if it exists, so default of None should be OK.